test/unit_tests/autoqasm/test_types.py

```diff
@@ -347,7 +347,7 @@ def main():
         annotation_test(1.0)
 
     expected = """OPENQASM 3.0;
-def annotation_test(float[64] input) {
+def annotation_test(float[64] input_) {
```

Comment:
What if a user tries to call this subroutine with a python kwarg? e.g.
```
    @aq.main
    def main():
        annotation_test(input=1.0)
```
We should properly translate the `input` kwarg to `input_` so that it matches the actual subroutine parameter.

Reply:
Should I change the `is_reserved_keyword` to return the new name as well?

Reply:
Thanks, this looks good!